draft-ietf-httpbis-cache-latest.xml

```diff
@@ -386,6 +386,34 @@
    prohibited from storing such responses.
 </t>
 
+<section title"Storing Header and Trailer Fields" anchor="storing.fields">
+<t>
+   Caches &MUST; include all received header fields -- particularly
+   unrecognised ones -- when storing a response; this assures that new HTTP
+   header fields can be successfully deployed. However, the following exceptions
+   are made:
+</t>
+<ul>
+   <li>Fields whose names are listed in the Connection header field are
+   required by <xref target="header.connection"/> to be removed before
+   forwarding the message. This &MAY; be implemented by doing so
+   before storage.</li>
+   <li>Likewise, fields that are known to be "hop-by-hop" are also required
+   to be removed before forwarding, and this &MAY; be implemented by doing so
+   before storage; see <xref target="header.connection"/> for some
+   examples.</li>
+   <li>Header fields that are specific to the connection they occur
+   &MUST-NOT; be stored. Effectively, this is limited to Proxy-Authenticate
+   <xref target="field.proxy-authenticate"/> and Proxy-Authorization <xref
+   target="field.proxy-authorization">, since new fields are
+   discouraged from being tied to a specific connection.</li>
```

Comment:
Note that this is dependent upon the resolution of #331, which may not go this way.